drivers/rtc/rtc-pcf8523.c

```diff
@@ -108,10 +108,10 @@ static int pcf8523_rtc_read_time(struct device *dev, struct rtc_time *tm)
 	if (err < 0)
 		return err;
 
-	if ((regs[0] & PCF8523_CONTROL1_STOP) || (regs[3] & PCF8523_SECONDS_OS))
+	if ((regs[0] & PCF8523_CONTROL1_STOP))
 		return -EINVAL;
 
-	if (regs[0] & PCF8523_SECONDS_OS) {
+	if (regs[3] & PCF8523_SECONDS_OS) {
```

Comment:
You could replace `3` with `PCF8523_REG_SECONDS` here and in the rest of this `if` block.